### PR TITLE
feat: yearn opportunities full multi-account support

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

- Overview
  - [x] Account selection
  - [x] Use selected account data
- [x] Deposit
  - [x] Approve
  - [x] Confirm
  - [x] Use selected account data in all steps
  - [ ] ~~Poll (regression test against account 0 on prod)~~ WS is super borked atm and Txs messages are not incoming
- [x] Withdraw
  - [ ] Approve
  - [x] Confirm 
  - [x] Use selected account data in all steps
  - [ ] ~~Poll (regression test against account 0 on prod)~~ WS is super borked atm and Txs messages are not incoming
 

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- relates to the closed https://github.com/shapeshift/web/issues/2605
- relates to https://github.com/shapeshift/web/issues/2615
- relates to https://github.com/shapeshift/web/issues/2730

## Risk

Tested with MetaMask (which has only account 0 support) but this could bring regressions for the Yearn DeFi modal for single accounts, test accordingly

## Testing (with KeepKey only currently)

- Add an Ethereum account and send some ETH and an asset to stake to it to it
- Stake in some opportunity, reload the app (websocket is currently borked) and click on the opportunity card for it
- Select account 1 and click deposit
- Notice account #1 is pre-selected at Deposit step, with the right amount to deposit 
- Deposit some more of your staked asset, sign and broadcast it - it should be deposited from account #1
- Disconnect and reconnect wallet, then reopen the overview modal and select withdraw with account #1 selected in overview
- Notice account #1 is pre-selected at Withdraw step, with the right amount available to withdraw
- Withdraw some of your staked asset - it should be withdrawn from account #1
- Disconnect and reconnect wallet, then reopen the Cosmos overview modal with account #1 selected

### Engineering

- Refer to top-level testing steps
- Polling logic looks sound as it's not able to be tested currently
- Double-check the logic looks sound in deposit withdraw and overview for all steps

### Operations

- Refer to top-level testing steps

## Screenshots (non-exhaustive, high gas prices for those Txs with KK currently, so I didn't screenshot the confirm step, note that with current ws issue it will never show as completed in the UI, but you can confirm completion on Etherscan)

image
image
image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2728** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2728" target="_blank">Graphite</a> 
    * **PR #2745** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2745" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2745?utm_source=stack-comment).

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Approve.tsx [hunk 1]
```diff
@@ -10,26 +11,32 @@ import type {
...
 
 type YearnApproveProps = {
+  accountId?: Nullable<AccountId>
   onNext: (arg: DefiStep) => void
 }
 
 const moduleLogger = logger.child({ namespace: ['YearnDeposit:Approve'] })
 
-export const Approve: React.FC<YearnApproveProps> = ({ onNext }) => {
+export const Approve: React.FC<YearnApproveProps> = ({ accountId, onNext }) => {
```
Comment (gomesalexandre) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Approve.tsx:39]: No inline

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Deposit.tsx [hunk 2]
```diff
@@ -20,18 +21,22 @@ import { logger } from 'lib/logger'
...
 import { DepositContext } from '../DepositContext'
 
 const moduleLogger = logger.child({ namespace: ['YearnDeposit:Deposit'] })
 
 export const Deposit: React.FC<
-  StepComponentProps & { onAccountIdChange: AccountDropdownProps['onChange'] }
-> = ({ onNext, onAccountIdChange: handleAccountIdChange }) => {
+  StepComponentProps & {
+    accountId?: Nullable<AccountId>
+    onAccountIdChange: AccountDropdownProps['onChange']
```
Comment (gomesalexandre) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Deposit.tsx:37]: No inline

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/Withdraw.tsx [hunk 3]
```diff
@@ -29,8 +31,11 @@ const moduleLogger = logger.child({
 })
 
 export const Withdraw: React.FC<
-  StepComponentProps & { onAccountIdChange: AccountDropdownProps['onChange'] }
-> = ({ onAccountIdChange: handleAccountIdChange, onNext }) => {
+  StepComponentProps & {
+    accountId: Nullable<AccountId>
+    onAccountIdChange: AccountDropdownProps['onChange']
```
Comment (gomesalexandre) [inline src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/Withdraw.tsx:36]: No inline

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/Status.tsx [hunk 4]
```diff
@@ -56,14 +57,13 @@ export const Status = () => {
...
-    selectFirstAccountSpecifierByChainId(state, chainId),
-  )
+  const accountAddress = useMemo(() => fromAccountId(accountId ?? '').account, [accountId])
 
   const serializedTxIndex = useMemo(() => {
-    if (!(state?.txid && state?.userAddress)) return ''
-    return serializeTxIndex(accountSpecifier, state.txid, state.userAddress)
-  }, [state?.txid, state?.userAddress, accountSpecifier])
+    if (!(state?.txid && accountId && accountAddress)) return ''
+    if (!(state?.txid && accountId && accountAddress?.length)) return ''
```
Comment (0xdef1cafe) [inline src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/Status.tsx:64]: do we need two lines here?

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/Status.tsx [hunk 5]
```diff
@@ -56,14 +57,13 @@ export const Status = () => {
...
+    return serializeTxIndex(accountId ?? '', state.txid, accountAddress)
```
Comment (0xdef1cafe) [inline src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/Status.tsx:65]:
```suggestion
    return serializeTxIndex(accountId, state.txid, accountAddress)
```

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Status.tsx [hunk 6]
```diff
@@ -47,14 +48,12 @@ export const Status = () => {
...
-    selectFirstAccountSpecifierByChainId(state, chainId),
-  )
+  const accountAddress = useMemo(() => fromAccountId(accountId ?? '').account, [accountId])
 
   const serializedTxIndex = useMemo(() => {
-    if (!(state?.txid && state?.userAddress)) return ''
-    return serializeTxIndex(accountSpecifier, state.txid, state.userAddress)
-  }, [state?.txid, state?.userAddress, accountSpecifier])
+    if (!(state?.txid && accountId && accountAddress?.length)) return ''
+    return serializeTxIndex(accountId ?? '', state.txid, accountAddress)
```
Comment (0xdef1cafe) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/components/Status.tsx:55]:
```suggestion
    return serializeTxIndex(accountId, state.txid, accountAddress)
```

**Review by 0xdef1cafe — APPROVED** [review]
approved w/ nit picks

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/Status.tsx (hunk 4 above)
Comment (gomesalexandre) [reply src/features/defi/providers/yearn/components/YearnManager/Withdraw/components/Status.tsx]: 512176076cd50f7548cfaa4a62bf1d92ba63e6f5

**Review by 0xdef1cafe — APPROVED** [review]

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
revert config?

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**